src/app/helpers/helperMethods.ts

```diff
@@ -32,4 +38,69 @@ export namespace helperMethods {
         }
         return false;
     };
+
+    export async function downloadProjectTemplate(projectFolder: string, projectRepo: string, projectBranch: string): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await request(`${projectRepo}/archive/${projectBranch}.zip`)
+                .pipe(fs.createWriteStream(zipFile))
+                .on('error', function () {
+                    throw new Error("unable to download project zip file")
```

Comment:
did you actually simulate the error case, e.g. using a path that doesn't exist? Can 'error' event provide the reason?
Please format the message as: `Unable to download the project zip file.` and use a semi-colon at the end of the line.

Reply:
I updated fs.createWriteStream and now it throws the following error (this is just an example, of course) when providing an invalid path 
Error: ENOENT: no such file or directory, open 'C:\generator-office\foo\project.zip'
Emitted 'error' event at:
    at lazyFs.open (internal/fs/streams.js:115:12)
    at FSReqWrap.oncomplete (fs.js:141:20)